### PR TITLE
Grpc upgrade w logs

### PR DESCRIPTION
**Pull request by sjezewski**
This PR:

- upgrades our grpc library (the current one is 7mos old)
- rips out pedge rpclog / lion / protolion libraries (there was a conflict / redefinition w the new grpc lib)

### CODE REVIEW

**Comment by sjezewski** [comment]
The old logs look like:

```
2017-06-12T21:59:58Z INFO  protorpclog.Call {"service":"pps.API","method":"InspectJob","request":"job:\u003cid:\"aa39e035-8fa0-4d87-8de8-99651c40958a\" \u003e ","duration":"0.000s"}
```

I tried to make the new log format similar to the old, the new log lines now look like:

```
2017-07-13T21:19:52Z INFO pfs.API.ListRepo {"duration":0,"error":null,"request":{},"response":null} 
2017-07-13T21:19:52Z INFO pfs.API.ListRepo {"duration":0.00038794,"error":null,"request":{},"response":{"repo_info":[{"repo":{"name":"ttt"},"created":{"seconds":1499906398,"nanos":434489310},"size_bytes":12826738}]}} 
```

But have less redundant info (that `protorpclog.Call` bit), put duration / error fields towards the beginning which I think makes scanning visually a bit easier), and don't double escape quotes ... so the fields are proper JSON, and so a bit easier to read.

**Comment by sjezewski** [comment]
One other thing to note ...

[This commit](https://github.com/pachyderm/pachyderm/pull/2038/commits/68cfd5ffeb27543173ddb615adad66248aadf9cd) shows the manual modification of vendored code I had to do.

As the commit message suggests, I think this is OK in this instance. There is a known issue w the upper/lower case versions of the log library [here](https://github.com/sirupsen/logrus/issues/570), and we can tell what deps we've modified manually by doing `govendor status`

File: src/server/pfs/server/driver.go [hunk 1]
```diff
@@ -327,15 +328,22 @@ func (d *driver) listRepo(ctx context.Context, provenance []*pfs.Repo) ([]*pfs.R
...
+	fmt.Printf("got repo iterator %v\n", err)
 	if err != nil {
 		return nil, err
 	}
 nextRepo:
 	for {
 		repoName, repoInfo := "", new(pfs.RepoInfo)
+		fmt.Printf("getting a repo %v\n", repoName)
 		ok, err := iterator.Next(&repoName, repoInfo)
+		fmt.Printf("next repo? %v, %v\n", ok, err)
```
Comment (jdoliner) [inline src/server/pfs/server/driver.go]: Probably want to remove these printfs.

Comment (sjezewski) [inline src/server/pfs/server/driver.go]: Ooops, yup

**Comment by peter-edge** [comment]
Just happened to come over here to see what's been up with Pachyderm - man so much new stuff, good work haha. Just note, feel free to ping me if you need something, ie gRPC being upgraded in lion. Also note, I would recommend moving to github.com/uber-go/zap, it's (IMO) the best logging lib out there, and I thought lion was before :-)

**Pull request merged**